packages/forms/src/model.ts

```diff
@@ -425,32 +425,118 @@ export abstract class AbstractControl {
 
   /**
    * Sets the synchronous validators that are active on this control.  Calling
-   * this overwrites any existing sync validators.
+   * this overwrites any existing synchronous validators.
+   *
+   * When you add or remove a validator at run time, you must call
+   * `updateValueAndValidity()` for the new validation to take effect.
+   *
+   * If you want to add a new validator without affecting existing ones, consider
+   * using `addValidators()` method instead.
+   */
+  setValidators(validators: ValidatorFn|ValidatorFn[]|null): void {
+    this._rawValidators = validators;
+    this._composedValidatorFn = coerceToValidator(validators);
+  }
+
+  /**
+   * Sets the asynchronous validators that are active on this control. Calling this
+   * overwrites any existing asynchronous validators.
+   *
+   * When you add or remove a validator at run time, you must call
+   * `updateValueAndValidity()` for the new validation to take effect.
+   *
+   * If you want to add a new validator without affecting existing ones, consider
+   * using `addAsyncValidators()` method instead.
+   */
+  setAsyncValidators(validators: AsyncValidatorFn|AsyncValidatorFn[]|null): void {
+    this._rawAsyncValidators = validators;
+    this._composedAsyncValidatorFn = coerceToAsyncValidator(validators);
+  }
+
+  /**
+   * Add a synchronous validator or validators to this control, without affecting other validators.
    *
    * When you add or remove a validator at run time, you must call
    * `updateValueAndValidity()` for the new validation to take effect.
    *
+   * Adding a validator that already exists will have no effect.
+   *
```

Comment:
May be add a quick note on the behavior that Pete pointed out (and similar note to the `addAsyncValidators` function):
```suggestion
   * Adding a validator that already exists will have no effect.
   *
   * If duplicate validator functions are present in the `validators` array, only the first instance
   * would be added to a form control.
```